GameServerLib/Logic/GameObjects/Unit.cs

```diff
@@ -739,6 +744,36 @@ public ClassifyUnit ClassifyTarget(Unit target)
             return ClassifyUnit.Default;
         }
 
+        public byte GetBuffSlot(Buff b)
+        {
+            foreach (var buff in BuffSlotToRemove)
+            {
+                if (!AppliedBuffs.Values.Contains(buff))
+                    continue;
+
+                for (byte i = 1; i <= AppliedBuffs.Count + 1; i++)
```

Comment:
```C#
foreach (var kvp in AppliedBuffs)
{
    if (kvp.Value.Equals(buff)) {
        AppliedBuffs.Remove(kvp.Key);
    }
}
```

Reply:
Can't use a foreach loop since it would modify the collection and crash on the next iteration.